### PR TITLE
chore/PSD-3525_gtm_cookie_investigation

### PR DESCRIPTION
**Pull request by tahir-khalid**
Ensured GTM script is firing at correct time and cookies_policy is being set.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -14,12 +14,12 @@
     <%= javascript_include_tag "application", type: "module", nonce: true %>
     <%= stylesheet_link_tag "application" %>
     <%= csrf_meta_tags %>
-    <% if Rails.env.production? && analytics_cookies_accepted? %>
+    <% if !Rails.env.development? %>
```
Comment (alan-at-work) [inline app/views/layouts/application.html.erb:17]: maybe use `<% unless Rails.env.development? %>` also why are we not checking the `analytics_cookies_accepted` now?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -14,12 +14,12 @@
...
       <%= render "shared/ga_head" %>
     <% end %>
   </head>
   <body class="govuk-template__body app-body-class">
-    <% if Rails.env.production? && analytics_cookies_accepted? %>
+    <% if !Rails.env.development? %>
```
Comment (alan-at-work) [inline app/views/layouts/application.html.erb:22]: as above maybe use `<% unless Rails.env.development? %>`

File: app/controllers/concerns/cookies_concern.rb [hunk 3]
```diff
@@ -49,7 +49,8 @@ def set_analytics_cookies(accept_analytics_cookies)
       cookies[:cookies_policy] = {
```
Comment (alan-at-work) [inline app/controllers/concerns/cookies_concern.rb:49]: Seems if you're not calling this is it really needed?

**Review by alan-at-work — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (tahir-khalid) [reply app/views/layouts/application.html.erb:17]: Reading John's email the issue seems to be the GTM container doesn't respond unless analytics cookie is accepted so the intention here is bypass that trigger the container regardless.

Comment (alan-at-work) [reply app/views/layouts/application.html.erb:17]: that means you're not setting the cookie, I think they need to remove that and trust that we are dealing with it or update their code to use the one we have set.

File: app/controllers/concerns/cookies_concern.rb (hunk 3 above)
Comment (tahir-khalid) [reply app/controllers/concerns/cookies_concern.rb:49]: Adjust code to ensure cookies policy is created.

**Comment by tahir-khalid** [comment]
No longer required resolved in GTM 360 @ DBT

**Comment by tahir-khalid** [comment]
Closing as this is not needed now they used another cookie to trigger the container.

**Pull request closed without merging**